docs/performances.md

```diff
@@ -0,0 +1,169 @@
+    // ... method implementation
+
+    this.performanceApi.measure({
+      markName: markQuery,
+      measureName: `${markName} success`,
+      category: 'CozyClient queries'
+    })
+  } catch (error) {
+    this.performanceApi.measure({
+      markName: markQuery,
+      measureName: `${markName} error`,
+      category: 'CozyClient queries',
+      color: 'error'
+    })
+  }
+}
+```
+
+## Adding measurements
+
+On initial implementation, we chose to measure some parts of the code that we estimated useful and with potential high impact on performances (i.e. queries, store management, pouch management etc.)
+
+We excpect to need more measurements in the future but we need to care to keep the balance beetween having useful measurements and to not visualy overload the measurement graphs
+
+The API is meant to be simple to use so measurements can be quickly added temporarily in development environements
+
+In the scenario where a measurement seems to be useful enough to be permanently added to our code base, please consider discussing this choice with the development team.
+
+## Web usage
+
+On web environement, it is possible to inject the already provided `webPerformanceAPI` implementation of the API
+
+Example of CozyClient initialization with web compatible version of the PerformanceAPI
+```js
+import CozyClient, { webPerformanceAPI } from 'cozy-client'
+
+const client = new CozyClient({
+  uri,
+  schema,
+  token,
+  performancesApi: webPerformanceAPI
+})
+```
+
+Then CozyClient will use this implementation to do its internal measurements
+
+The consuming app can also add its own measurements by either:
+- Calling `client.performanceAPI.mark()` and `client.performanceAPI.measure()` method
+- Directly calling the web native API `performance.mark()` and `performance.measure()`
+  - However, please keep in mind that if both APIs will produce the same output, their interface is slightly different. So for code homogeneity purpose, please prefer using the CozyClient's API
+
+When the PerformanceAPI is used, then the result can be visualized in the browser's Performance develools
+
+![Performance devtools screenshot](./performances_devtools.png)
+
+## Other platforms
+
+No implementation is provided by CozyClient for other platforms. So each consuming app needs to provide their own implementation of the API
+
+The API interface is described in the JSDoc
+
+Example of custom implementations:
+- Flagship App: [](TODO ADD LINK)
+
+## Production environment
+
+By default, if no API is injected, then CozyClient will use the `defaultPerformanceAPI` implementation that does nothing
+
+The consuming app is then responsible to inject an implementation or keep the default one that doesn't do any profiling, regarding if the app is built for development environments or for production
+
+PerformancesAPI seems to have an impact on performances light enough to be kept on production. However, we suggest to disable them in that environement if possible
```

Comment:
s/environement/environment

Reply:
> light enough to be kept on production

Are we sure of this? I guess it's ok for a single measure, but I would assume a significative impact overall, since there can be a lot of measures during an app data lifecycle

Reply:
My guess is that, because the API can measure durations as small as a few μs, on high level methods this should not be perceptible. But this may be different on some API for other platforms. Also I did not find any info about it so this is just a guess.

But that's why I suggested to disable it.